### ISSUE TITLE
Table.insertRow insert duplicates and cause ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by ShadowNinja44** [processing/processing#5406]
## Description

`table.insertRow(int, Object[])` inserts the given Object[] two times in the table but does not seem to set certain variables correctly. Attempting to access the last element (I think) will give an ArrayIndexOutOfBoundsException

## Example / Test code
```processing
void setup() {
    println("******************************");
    test1();
    println("******************************");
    test2();
    println("******************************");
}
void test1() {
    Table t = new Table();
    println("Rows: " + t.getRowCount());
    t.addRow(new Object[0]);
    println("Rows: " + t.getRowCount());
    t.print();
    t.removeRow(0);
    println("Rows: " + t.getRowCount());
    t.print();
    t.insertRow(0, new Object[0]);
    println("Rows: " + t.getRowCount());
    t.print();
    t.removeRow(0);
    println("Rows: " + t.getRowCount());
    t.print();
}
void test2() {
    Table t = new Table();
    println("Rows: " + t.getRowCount());
    t.addRow(new Object[]{"test1", "test2"});
    println("Rows: " + t.getRowCount());
    t.print();
    t.removeRow(0);
    println("Rows: " + t.getRowCount());
    t.print();
    t.insertRow(0, new Object[]{"test1", "test2"});
    println("Rows: " + t.getRowCount());
    t.print();
    t.removeRow(0);
    println("Rows: " + t.getRowCount());
    t.print();
}
```

### Example / Test code (expected) output
```
******************************
Rows: 0
Rows: 1

Rows: 0
Rows: 1

Rows: 0
******************************
Rows: 0
Rows: 1
test1	test2
Rows: 0
Rows: 1
test1	test2
Rows: 0
******************************
```

### Example / Test code (current) output
```
******************************
Rows: 0
Rows: 1

Rows: 0
Rows: 2


Rows: 1

******************************
Rows: 0
Rows: 1
test1	test2
Rows: 0
Rows: 2
# Here the ArrayIndexOutOfBoundsException is caused. Line 35 in the code.
```

## Steps to Reproduce

1. See test code above.

## Your Environment

 

* Processing version: 3.3.6
* Operating System and OS version: Ubuntu 17.10 & Windows 10 on separate machine.
* Other information:

## Possible Causes / Solutions

I've been going through the source code for Processing Table / TableRow but haven't been able to figure out the cause. It is possible that there is an "off by one" error somewhere or that something that shouldn't be called multiple times is called more than once.

**Comment by MLanghof** [comment]
The problem is that [`insertRow()` calls `setRow()` before actually increasing rowCount](https://github.com/processing/processing/blob/master/core/src/processing/data/Table.java#L2431). This means that the [ensureBounds](https://github.com/processing/processing/blob/master/core/src/processing/data/Table.java#L2550) increases `rowCount` [here](https://github.com/processing/processing/blob/master/core/src/processing/data/Table.java#L4159) once, then upon returning to `insertRow()` it is incremented again. The table is then in an inconsistent state and `print()` (or most other functions) fail.

**Comment by benfry** [comment]
Now fixed for 3.3.7.

@MLanghof Thanks 100x for your help. Again, can't tell you how much it's appreciated.
